ipdb/__main__.py

```diff
@@ -12,66 +12,39 @@
 
 __version__= "0.10.3"
 
-def import_module(possible_modules, needed_module):
-    """Make it more resilient to different versions of IPython and try to
-    find a module."""
-    count = len(possible_modules)
-    for module in possible_modules:
-        try:
-            return __import__(module, fromlist=[needed_module])
-        except ImportError:
-            count -= 1
-            if count == 0:
-                raise
-try:
-    # IPython 5.0 and newer
-    from IPython.terminal.debugger import TerminalPdb as Pdb
-    from IPython.core.debugger import BdbQuit_excepthook
-    from IPython.terminal.interactiveshell import TerminalInteractiveShell
-    # Let IPython decide about which debugger class to use
-    # This is especially important for tools that fiddle with stdout
-    debugger_cls = TerminalInteractiveShell().debugger_cls
-except ImportError:
-    from IPython.core.debugger import Pdb, BdbQuit_excepthook
-    debugger_cls = Pdb
-
-possible_modules = ['IPython.terminal.ipapp',           # Newer IPython
-                    'IPython.frontend.terminal.ipapp']  # Older IPython
-
-app = import_module(possible_modules, "TerminalIPythonApp")
-TerminalIPythonApp = app.TerminalIPythonApp
-
-possible_modules = ['IPython.terminal.embed',           # Newer IPython
-                    'IPython.frontend.terminal.embed']  # Older IPython
-embed = import_module(possible_modules, "InteractiveShellEmbed")
-InteractiveShellEmbed = embed.InteractiveShellEmbed
-try:
-    get_ipython
-except NameError:
+from IPython import get_ipython
+from IPython.core.debugger import BdbQuit_excepthook
+from IPython.terminal.ipapp import TerminalIPythonApp
+from IPython.terminal.embed import InteractiveShellEmbed
+
+
+shell = get_ipython()
+if shell is None:
+    # Not inside IPython
     # Build a terminal app in order to force ipython to load the
     # configuration
     ipapp = TerminalIPythonApp()
     # Avoid output (banner, prints)
     ipapp.interact = False
     ipapp.initialize([])
+    # Let IPython decide about which debugger class to use
+    # This is especially important for tools that fiddle with stdout
+    debugger_cls = ipapp.shell.debugger_cls
     def_colors = ipapp.shell.colors
+    def_exec_lines = [line + '\n' for line in ipapp.exec_lines]
```

Comment:
@gotcha I don't think using this makes sense: the lines are passed to `Pdb.rcLines` below, but that appears to be meant for pdb commands, not Python code. And `ipapp` will have already executed the code here as part of `initialize([])` above. These bugs are probably hidden because very few people set `exec_lines`.

So I'm inclined to get rid of this `def_exec_lines` connection entirely, but am I missing something? Is there an important reason this is needed?

Reply:
I found PR #69, where it was added, and issue #61. Neither of them mention this specific bit of config, so I'm going to assume it was a mistake, and cut it out.